fbu_twitter_starter/twitter/Models/Tweet.m

```diff
@@ -0,0 +1,57 @@
+//
+//  Tweet.m
+//  twitter
+//
+//  Created by Abigail Shilts on 6/20/22.
+//  Copyright © 2022 Emerson Malca. All rights reserved.
+//
+
+#import "DateTools.h"
+#import "stringsList.h"
+#import "Tweet.h"
+#import "User.h"
+
+@implementation Tweet
+- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
+    self = [super init];
+
+    if (self) {
+        NSDictionary *originalTweet = dictionary[retweetStatus];
```

Comment:
Not necessary in this assignment, but in future keep in mind if any of these properties are missing in the dictionary you can't create your Tweet object. Because Tweet class properties are marked as non_null in your code by NS_ASSUME_NONNULL_BEGIN/NS_ASSUME_NONNULL_END directives.

In real-world app you want to check against property presence in the dictionary and fail if it's missing. Instead of using `initWithDictionary` initializer it's better to have a static method that creates Tweet object from the dictionary and returns error if required property is missing.

`+ (nullable Tweet*) tweetFromJson:(NSDictionary *)dictionary error: (NSError**)error;`